hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableMetaClient.java

```diff
@@ -244,7 +244,7 @@ public void updateFunctionalIndexMetadata(HoodieFunctionalIndexMetadata newFunct
 
   public static HoodieTableMetaClient reload(HoodieTableMetaClient oldMetaClient) {
     return HoodieTableMetaClient.builder()
-        .setConf(oldMetaClient.storageConf.newInstance())
+        .setStorage(oldMetaClient.getStorage())
```

Comment:
don't we need a clone/create a new copy here?

Reply:
Given this is reloading, there is no real necessity to make a copy of the conf or storage.